stdlib/public/core/ManagedBuffer.swift

```diff
@@ -106,8 +106,10 @@ extension ManagedBuffer {
   /// Call `body` with an `UnsafeMutablePointer` to the stored
   /// `Header`.
   ///
-  /// - Note: This pointer is valid only for the duration of the
-  ///   call to `body`.
+  /// - Note: This pointer is fixed when the buffer is created, but it is only valid
+  ///   for the lifetime of the buffer. The buffer is guaranteed to live for the
+  ///   duration of the call to `body`, but may be deallocated immediately thereafter
+  ///   if its lifetime is not otherwise extended.
```

Comment:
```suggestion
  /// - Note: This pointer is valid only for the duration of the
  ///   call to `body`.
  ///   However, the pointer's address doesn't change
  ///   during the lifetime of the buffer,
  ///   which lets you pass the pointer to C functions
  ///   that expect a stable pointer,
  ///   even if those function calls are made in separate calls to
  ///   `withUnsafeMutablePointerToHeader(body:)`
```

Reply:
C APIs are just one example of APIs that may rely on the stability of pointers. Swift has plenty of Unsafe APIs with the same behavior. And `os_unfair_lock` is a Darwin-only concept; it doesn’t exist on other platforms that Swift runs on, like Linux or Windows.

Reply:
Revised suggestion:

```suggestion
  /// This pointer is valid only for the duration of the
  /// call to `body`.
  /// If you try to access the pointer outside of body,
  /// the buffer might have already been deallocated, causing a crash.
  /// However, the pointer's address doesn't change
  /// during the lifetime of the buffer,
  /// which lets you pass the pointer to functions
  /// that expect a stable pointer,
  /// even if those function calls are made in separate calls to
  /// `withUnsafeMutablePointerToHeader(body:)`
```